sendbuffer.go

```diff
@@ -100,17 +104,41 @@ func (buf *sendBuffer) sendLoop(w io.Writer) {
 	}
 }
 
+func (buf *sendBuffer) send(b []byte, writeDeadline time.Time) (int, error) {
+	buf.muClosing.RLock()
+	defer buf.muClosing.RUnlock()
+	if buf.closing {
```

Comment:
Here is the main change. The code is moved from `stream.go`. `muClosing` is to prevent sending to `buf.in` if it's closed.